akka-sample-cqrs-scala/src/main/scala/sample/cqrs/CqrsApp.scala

```diff
@@ -42,24 +42,34 @@ object CqrsApp {
   }
 
   def startNode(port: Int): Unit = {
+
     val system = ActorSystem("ClusterSystem", config(port))
 
     // TODO start ClusterSharding
 
-    if (port == 2551) {
+    if (port <= 2551) {
       createTables(system)
 
       testIt(system)
     }
 
+    if (port > 2551)
+      EventProcessorWrapper(system).start()
```

Comment:
This must be done on all nodes, since it's initializing sharding and the singleton

Reply:
instead of looking at the port you can look at the role with `Cluster(system).selfRoles`, or put that inside the EventProcessorWrapper.start method